src/api/status/src/js/github-stats.js

```diff
@@ -1,74 +1,124 @@
 const { fetch } = require('@senecacdot/satellite');
 
+const redis = require('../redis');
+
 const fetchGitHubApi = (owner, repo, path) =>
   fetch(`https://api.github.com/repos/${owner}/${repo}/${path}`, {
     headers: { Accept: 'application/vnd.github.v3+json' },
   });
 
-module.exports = async function getGitHubData(owner, repo) {
-  const returnedValue = {};
+const statsParticipation = async (owner, repo) => {
+  // Get data from cached data
+  const cached = await redis.get(`github:statsParticipation-${repo}`);
+  if (cached) return JSON.parse(cached);
 
   // get weekly commits for last year: https://docs.github.com/en/rest/reference/repos#get-the-weekly-commit-count
-  try {
-    const res = await fetchGitHubApi(owner, repo, 'stats/participation');
-    const participationResponse = await res.json();
-
-    if (!res.ok) throw new Error(`[Error ${res.status}] - ${participationResponse.message}`);
-
-    returnedValue.weeklyCommits = {
-      commits: participationResponse.all[participationResponse.all.length - 1],
-    };
-    returnedValue.yearlyCommits = {
-      commits: participationResponse.all.reduce((a, b) => a + b),
-    };
-  } catch (error) {
-    console.error(error);
-  }
+  const res = await fetchGitHubApi(owner, repo, 'stats/participation');
+  const participationResponse = await res.json();
+  const returnedValue = {};
+  if (!res.ok) throw new Error(`[Code ${res.status}] - ${participationResponse.message}`);
+
+  returnedValue.weeklyCommits = {
+    commits: participationResponse.all[participationResponse.all.length - 1],
+  };
+  returnedValue.yearlyCommits = {
+    commits: participationResponse.all.reduce((a, b) => a + b),
+  };
+
+  await redis.set(
+    `github:statsParticipation-${repo}`,
+    JSON.stringify(returnedValue),
+    'EX',
+    10 * 60
```

Comment:
No magic numbers.  Please create a variable for this that specifies what it is, and the units (seconds).